src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -220,22 +222,45 @@ public void setRatedIfNotExceeded(@Nullable ZonedDateTime exerciseDueDate, Zoned
      * - no due date is set OR
      * - the submission type is INSTRUCTOR / TEST
      *
-     * @param exerciseDueDate date after which no normal submission is considered rated.
-     * @param submission      to which the result belongs.
-     * @param participation   to wich the submission belongs
+     * @param dueDate    due date of the exercise or individual due date if applicable
+     * @param submission to which the result belongs.
      */
-    public void setRatedIfNotExceeded(ZonedDateTime exerciseDueDate, Submission submission, Participation participation) {
+    public void setRatedIfNotAfterDueDate(@Nullable ZonedDateTime dueDate, @NotNull Submission submission) {
         if (submission.getType() == SubmissionType.INSTRUCTOR || submission.getType() == SubmissionType.TEST) {
             this.rated = true;
         }
         else if (submission.getType() == SubmissionType.ILLEGAL || participation.isTestRun()) {
             this.rated = false;
         }
         else {
-            setRatedIfNotExceeded(exerciseDueDate, submission.getSubmissionDate());
+            setRatedIfNotAfterDueDate(dueDate, submission.getSubmissionDate());
         }
     }
 
+    /**
+     * Sets the rated attribute depending on the submission type and date.
+     * This method takes the individual due date specified by the participation attribute into account
+     *
+     * @param submission    to which the result belongs.
+     * @param participation to wich the submission belongs
+     */
+    public void setRatedIfNotAfterDueDate(@NotNull Submission submission, @NotNull Participation participation) {
+        ZonedDateTime dueDate = ExerciseDateService.getDueDate(participation).orElse(null);
+        setRatedIfNotAfterDueDate(dueDate, submission);
+    }
```

Comment:
If you don't need the submission + participation above, you can also remove them here.